readthedocs/oauth/services/github.py

```diff
@@ -97,21 +100,19 @@ def create_repository(self, fields, privacy=None, organization=None):
         """
         privacy = privacy or settings.DEFAULT_PRIVACY_LEVEL
         if any([
-                (privacy == 'private'),
-                (fields['private'] is False and privacy == 'public'),
+            (privacy == 'private'),
+            (fields['private'] is False and privacy == 'public'),
         ]):
-            try:
-                repo = RemoteRepository.objects.get(
-                    full_name=fields['full_name'],
-                    users=self.user,
-                    account=self.account,
-                )
-            except RemoteRepository.DoesNotExist:
-                repo = RemoteRepository.objects.create(
-                    full_name=fields['full_name'],
-                    account=self.account,
-                )
-                repo.users.add(self.user)
+
+            repo, _ = RemoteRepository.objects.get_or_create(
+                full_name=fields['full_name']
+            )
+            remote_relation, _ = RemoteRepositoryRelation.objects.get_or_create(
+                remoterepository=repo,
+                user=self.user,
+                account=self.account
```

Comment:
This might fail if the user has multiple connected account and more than one of them has access to the same repo, we current have unique_together(``remoterepository``, ``user``) so this will raise ``IntegrityErrot`` while trying to create a relation with the same user and remore_repo but with different account,

So, either we need to allow only one connected account for a user to have access to the same repo or have different logic around the uniqueness.

Though this is an edge case

Reply:
@humitos  Just tested it, Users can add multiple GitHub accounts to their profile and if both of them have access to the same GitHub Repository then it will cause the error.

This is the error user will get:

``There was an error with your request: UNIQUE constraint failed: oauth_remoterepositoryrelation.remoterepository_id, oauth_remoterepositoryrelation.user_id``

Let me know your thought on this :)

Reply:
Hrm... So, I see some options here:

- remove the unique constraint here
- change it to be `unique(remoterepository_id, socialaccount_id)` instead

The second one seems more accurate to me since we don't really need to filter by user since the social account is already connected to a user.

Reply:
Yes, I have updated the PR with the second option, now the error I mentioned above has gone away :)